src/audio/host-zephyr.c

```diff
@@ -836,8 +837,9 @@ int host_common_params(struct host_data *hd, struct comp_dev *dev,
 	}
 
 	/* calculate DMA buffer size */
-	buffer_size = ALIGN_UP(period_bytes, align) * period_count;
-	buffer_size = MAX(buffer_size, ALIGN_UP(hd->ipc_host.dma_buffer_size, align));
+	round_up_size = (params->frame_fmt == SOF_IPC_FRAME_S24_3LE) ? (3 * align) : align;
+	buffer_size = ROUND_UP(period_bytes, round_up_size) * period_count;
+	buffer_size = MAX(buffer_size, ROUND_UP(hd->ipc_host.dma_buffer_size, buffer_size));
```

Comment:
the last line doesn't look right. Comparing to the current code it should be
```
buffer_size = MAX(buffer_size, ROUND_UP(hd->ipc_host.dma_buffer_size, round_up_size));
```
but if you really want to round up to a multiple of `buffer_size` then you don't need a `MAX()` - the rounding result won't be smaller, as long as `dma_buffer_size != 0`